### PR TITLE
openMP update and PDOS+soc

### PR DESCRIPTION
**Pull request by dyzheng**

### CODE REVIEW

**Review by wenfei-li — APPROVED** [review]

File: source/module_orbital/ORB_atomic_lm.cpp [hunk 1]
```diff
@@ -224,7 +227,7 @@ void Numerical_Orbital_Lm::extra_uniform(const double &dr_uniform_in, const bool
 
 	// do interpolation here to make grid more dense
 
-#ifdef __OPENMP
+#ifdef _OPENMP
 	#pragma omp parallel for schedule(static)
 #endif
```
Comment (caic99) [inline source/module_orbital/ORB_atomic_lm.cpp:232]: Where's the original corresponding `#ifdef` of this `#endif`? It seems that only `ifdef`s are added in this commit.

File: source/module_orbital/ORB_atomic_lm.cpp [hunk 2]
```diff
@@ -266,10 +269,10 @@ void Numerical_Orbital_Lm::extra_uniform(const double &dr_uniform_in, const bool
...
 		case 4: ModuleBase::Mathzone_Add1::SplineD2 (ModuleBase::GlobalFunc::VECTOR_TO_PTR(r_radial), ModuleBase::GlobalFunc::VECTOR_TO_PTR(psi), nr, 0.0, 0.0, y2); break;
 		default: 
-			//GlobalV::ofs_warning << " The angular momentum larger than 4 (g orbitals) may be error about eggbox. " << std::endl;
-			//GlobalV::ofs_warning << " Check file " << __FILE__ << " line " << __LINE__ <<std::endl;
-			std::cout << " The angular momentum larger than 4 (g orbitals) may be error about eggbox. " << std::endl;
-			std::cout << " Check file " << __FILE__ << " line " << __LINE__ <<std::endl;
+			GlobalV::ofs_warning << " The angular momentum larger than 4 (g orbitals) may be error about eggbox. " << std::endl;
+			GlobalV::ofs_warning << " Check file " << __FILE__ << " line " << __LINE__ <<std::endl;
+			//std::cout << " The angular momentum larger than 4 (g orbitals) may be error about eggbox. " << std::endl;
+			//std::cout << " Check file " << __FILE__ << " line " << __LINE__ <<std::endl;
```
Comment (caic99) [inline source/module_orbital/ORB_atomic_lm.cpp:275]: I suggest removing this redundant commented code.

Comment (dyzheng) [reply source/module_orbital/ORB_atomic_lm.cpp:275]: I agree

File: source/module_orbital/ORB_atomic_lm.cpp (hunk 1 above)
Comment (dyzheng) [reply source/module_orbital/ORB_atomic_lm.cpp:232]: linpz said that _OPENMP would be automaticly setup by openmp setup in Makefile

**Pull request merged**